src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Listeners/ServiceBusListener.cs

```diff
@@ -2,42 +2,65 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Generic;
+using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
+using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Azure.ServiceBus;
 using Microsoft.Azure.ServiceBus.Core;
+using Microsoft.Azure.ServiceBus.Management;
 using Microsoft.Azure.WebJobs.Host.Executors;
 using Microsoft.Azure.WebJobs.Host.Listeners;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Extensions.Logging;
 
 namespace Microsoft.Azure.WebJobs.ServiceBus.Listeners
 {
-    internal sealed class ServiceBusListener : IListener
+    internal sealed class ServiceBusListener : IListener, IScaleMonitor<ServiceBusTriggerMetrics>
     {
-        private readonly MessagingProvider _messagingProvider;
+        private const string DeadLetterQueuePath = @"/$DeadLetterQueue";
+
+        private readonly MessagingProvider _messagingProvider;
+        private readonly string _functionId;
+        private readonly EntityType _entityType;
         private readonly string _entityPath;
-        private readonly bool _isSessionsEnabled;
+        private readonly bool _isSessionsEnabled;
+        private readonly bool _isListeningOnDeadLetterQueue;
         private readonly ServiceBusTriggerExecutor _triggerExecutor;
         private readonly CancellationTokenSource _cancellationTokenSource;
         private readonly MessageProcessor _messageProcessor;
         private readonly ServiceBusAccount _serviceBusAccount;
-        private readonly ServiceBusOptions _serviceBusOptions;
-        private MessageReceiver _receiver;
-        private ClientEntity _clientEntity;
+        private readonly ServiceBusOptions _serviceBusOptions;
+        private readonly ScaleMonitorDescriptor _scaleMonitorDescriptor;
+        private readonly ILogger<ServiceBusListener> _logger;
+
+        private MessageReceiver _receiver;
+        private MessageReceiver _metricsReceiver;
```

Comment:
Do we need a separate receiver for metrics? If we change the existing Receiver property to do lazy create, we could reuse it for both purposes (both metrics and main listener)?